modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainer.java

```diff
@@ -36,13 +37,20 @@ public MySQLContainer(String dockerImageName) {
 
     @Override
     protected void configure() {
-        optionallyMapResourceParameterAsVolume(MY_CNF_CONFIG_OVERRIDE_PARAM_NAME, "/etc/mysql/conf.d", "mysql-default-conf");
+        optionallyMapResourceParameterAsVolume(MY_CNF_CONFIG_OVERRIDE_PARAM_NAME, "/etc/mysql/conf.d",
+                "mysql-default-conf");
 
         addExposedPort(MYSQL_PORT);
         addEnv("MYSQL_DATABASE", databaseName);
         addEnv("MYSQL_USER", username);
-        addEnv("MYSQL_PASSWORD", password);
-        addEnv("MYSQL_ROOT_PASSWORD", password);
+        if (password != null && !password.isEmpty()) {
+            addEnv("MYSQL_PASSWORD", password);
+            addEnv("MYSQL_ROOT_PASSWORD", password);
+        } else if (MYSQL_ROOT_USER.equalsIgnoreCase(username)) {
+            addEnv("MYSQL_ALLOW_EMPTY_PASSWORD", "yes");
+        } else {
+            throw new ContainerLaunchException("Empty password can be used only with the root user");
```

Comment:
I just found out, that we catch all exceptions and throw `ContainerLaunchException` in `GenericContainer`'s `doStart()` method, so it's really not that important which exception we throw here, but `ContainerLaunchException` makes sense of course.

Reply:
@kiview: should we check the cause of the exception to make sure that the ContainerLaunchException is thrown by this method  instead of others

Reply:
I was thinking something like this at first also. But in the end, we just want to check the public API and the container behaves as expected if we receive this exception, so I'd be good with it.